### PR TITLE
fix(rxExample): fix performance lag/delay

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-942

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM

### CODE REVIEW

**Comment by CITguy** [comment]
## rxCheckbox Integration Test
_Loads the Elements > Forms page (with numerous examples)_

### Before (2.9 sec delay)
screen shot 2016-06-24 at 2 04 56 pm

### After (no delay)
screen shot 2016-06-24 at 2 28 54 pm

File: demo/scripts/rxExample.js [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
+            return lineCount > MAX_PREVIEW_LINES;
+        }
     }//isExpandable()
 
+    function isExpandedOnLoad (content) {
+        if (!content) { return false; }
+
+        var lineCount = countLines(content);
+
+        return lineCount <= (MAX_PREVIEW_LINES + 1);
```
Comment (CITguy) [inline demo/scripts/rxExample.js]: styles show half of line 11 by default. if the line count **is** 11, expand by default and disable collapsing

## before (half of line 11 shown)
screen shot 2016-06-24 at 2 51 43 pm

## after (lines 1-11 all shown)
screen shot 2016-06-24 at 2 51 49 pm

File: demo/scripts/rxExample.js [hunk 2]
```diff
@@ -1,29 +1,54 @@
...
-        var lineCount = content.trim().split('\n').length;
-        return lineCount > MAX_PREVIEW_LINES;
+        if (!content) {
+            return false;
+        }
+        var lineCount = countLines(content);
+
+        if (lineCount === MAX_PREVIEW_LINES + 1) {
+            // collapse is unnecessary
+            return false;
```
Comment (CITguy) [inline demo/scripts/rxExample.js]: disable expand/collapse if exactly 11 lines (will always show all 11 lines)

File: demo/scripts/rxExample.js [hunk 3]
```diff
@@ -1,29 +1,54 @@
...
         scope: {
             name: '@'
         },
         link: function (scope) {
-            var baseName = 'examples/' + scope.name;
-            var _canExpand = { markup: false, js: false, less: false };
-            var _isExpanded = { markup: false, js: false, less: false };
-            // markup not included b/c it's required for minimum functionality
-            var _hasSource = { js: false, less: false };
+            scope.markupUrl = 'examples/' + scope.name + '.html'; // used by ng-include
```
Comment (CITguy) [inline demo/scripts/rxExample.js]: still necessary for use by `ng-include` of interactive demo

**Comment by 100stacks** [comment]
@CITguy, is this fix strictly for `2.x`, since we also have the updated `<rx-example>` in `1.x`?

**Comment by CITguy** [comment]
@100stacks, this is for 2.x because the upcoming 2.0.0 release will replace existing documentation at http://rackerlabs.github.io/encore-ui/. It won't matter in 1.x unless the documentation is built locally or for Travis/Codeship tests of 1.50.x-LTS (which should be few and far between).

**Comment by 100stacks** [comment]
LGTM

File: grunt-tasks/examples.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+var _ = require('lodash');
+
+module.exports = function (grunt) {
+    // Group assets with their example name
+    // Should result in something like:
+    //  'foo.bar': [
+    //      '/path/to/foo.bar.html',
+    //      '/path/to/foo.bar.js',
+    //      '/path/to/foo.bar.less'
+    //  ]
```
Comment (CITguy) [inline grunt-tasks/examples.js]: NOTE to self: document params to this function

**Comment by CITguy** [comment]
Grunt task documentation updated. Ready for review.

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**